Chat sky mp

rrov opened the pull request:
resolve #1
Доделываю:

Исчезновение при скроле.
Одноразовая анимация появления сообщения

Commit:
chat

Доделать:
1. Исчезновение при скроле
2. Цвета авторам

Commit:
chat

Commit:
chat

Commit:
chat

Commit:
chat

Commit:
chat

Добавлены рандомные цвета ников игроков, которые пишут в чат

Commit:
chat

Commit:
chat

Переход после удаления на общий чат

Commit:
Add Chat

Commit:
chat

Commit:
add

Commit:
chat

Сделал одиночное появление сообщения

Commit:
chat

Сделал отдаление записей у верха вьюпорта чата.

Commit:
chat

убрал лишнее

Commit:
Update chat.js

Commit:
Update chat.js

Commit:
Update chat.js

Commit:
Update chat.css

Commit:
new svg

Добавил отредактированный в Vectr svg файл, увеличил его так, чтобы еще входило смещение при появлении второй полосы менюхи.

Commit:
Update chat.css

Pospelove commented:
@Exhorts  Благодарю за проделанную работу. Перед тем, как я сделаю merge, пожалуйста, исправьте следующие проблемы:

1. Анимация отдаления сообщений должна выглядеть, как показано [здесь](https://user-images.githubusercontent.com/37947786/56676208-1560de00-66be-11e9-8fda-13abefa502d8.png).
2. Если это возможно, исправить [выпирающие чёрные углы](https://user-images.githubusercontent.com/37947786/56676509-bf406a80-66be-11e9-819d-a805d0e65f43.png).
3. Выделенные вкладки чата не должны [заходить за линию](https://user-images.githubusercontent.com/37947786/56676625-f9117100-66be-11e9-9753-eb2d4aafb3dd.png).
4. Необходимо скроллить чат вниз при появлении нового сообщения в любой из вкладок, кроме `Общий чат`.
5. Необходимо сделать поле ввода менее прозрачным, чем весь остальной чат. [См. здесь](https://user-images.githubusercontent.com/37947786/56676858-7b9a3080-66bf-11e9-94a4-4792bffba95b.png).
6. Табуляция в `chat.js` должна быть корректной. Исправить табуляцию быстро можно [тут](https://beautifier.io/). Обратите внимание, что нужно 2 пробела вместо 1 таба (опция Ident with 2 spaces).

Commit:
Chat

1. Сделано.
2. Надо менять svg-фон, либо делать собственный фон с svg углами (заплатки в виде полос при разных мониторах могут выдавать либо пробелы (белы линии) между основным фоном и полосой, либо наоборот наложение прозрачных объектов, что = более темной полосе.
3. Сделано.
4. Сделано.
5. Сделано.
6. Сделано.

Commit:
chat

Commit:
chat

Лучше видно анимацию на однострочных сообщениях

Pospelove commented:
@Exhorts 
При появлении нового сообщения чат всё ещё не скроллится вниз автоматически.
Анимация отдаления сообщений должна начинаться со 2 сообщения сверху (подчеркнул) 
![image](https://user-images.githubusercontent.com/37947786/56684458-1438ac80-66d0-11e9-8079-99a0c4a2aaa9.png)

Также, пожалуйста, удалите `front/js/chat-oldcode.js`.

Commit:
Delete chat-oldcode.js

Commit:
Update chat.js

Скрол починил

Commit:
Update chat.js

Commit:
Update chat.js

Commit:
Scroll animation

Переделал анимацию исчезновения сообщений

Commit:
chat

Commit:
chat

Commit:
Update chat.css

Commit:
chat

захватывает 3 сообщения

Pospelove commented:
@Exhorts 
Всё отлично работает. Единственное, что необходимо - сделать, чтобы анимация отдаления не проигрывалась, если все сообщения влезают на страницу чата. Иначе получается такое:
![image](https://user-images.githubusercontent.com/37947786/56738910-e9049a80-676d-11e9-85ea-904a83bb5b17.png)

Commit:
Замена svg

Commit:
Update chat.js

Pospelove commented:
@Exhorts Благодарю за выполненную работу!
